### PR TITLE
doc: add dcerpc

### PR DESCRIPTION
**Pull request by travisbgreen**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3029

Describe changes:
- incorporated feedback from PR https://github.com/OISF/suricata/pull/4117

### CODE REVIEW

File: doc/userguide/rules/dcerpc-keywords.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+directly in UDP or within SMB via the dcerpc application layer protocol. Because
+of this we are able to detect within DCE/RPC the interface UUIDs, operation
+numbers, and for bytes within the stub data. This is achieved by using the
+keywords ``dcerpc.iface`` and ``dcerpc.opnum``, and the sticky buffer
+``dcerpc.stub_data``.
+
+
+dcerpc.iface
+------------
+It is necessary for a DCE/RPC client to bind to a service before being able call
```
Comment (norg) [inline doc/userguide/rules/dcerpc-keywords.rst]: being able to call it?

**Comment by norg** [comment]
LGTM, just a minor thing I found

File: doc/userguide/rules/dcerpc-keywords.rst (hunk 1 above)
Comment (travisbgreen) [reply doc/userguide/rules/dcerpc-keywords.rst]: Hmmm... maybe s/service/interface/?
"An interface is a set of remotely callable operations offered by a server and invokable by clients."
http://pubs.opengroup.org/onlinepubs/009629399/chap2.htm#tagcjh_05_01_01_01

File: doc/userguide/rules/dcerpc-keywords.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  dcerpc.opnum:19;
+  dcerpc.opnum:12-14,12,121,62-78;
+  dcerpc.opnum:12,26,62,61,6513-6666;
+
+Siganture Example:
+
+.. container:: example-rule
+
+  alert tcp any any -> $HOME_NET any (msg:"TGI LATERAL DCE/RPC Service Control Manager Interface UUID with StartServiceW Operation Number"; :example-rule-emphasis:`dcerpc.iface:367abb81-9844-35f1-ad32-98f038001003;` dcerpc.opnum:19; sid:1;)
+
```
Comment (victorjulien) [inline doc/userguide/rules/dcerpc-keywords.rst]: Missing reference to old notation here.

File: doc/userguide/rules/dcerpc-keywords.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+dcerpc.stub_data
+----------------
+``dcerpc.stub_data`` is a sticky buffer that points to the buffer containing 
+the DCE/RPC request or response stub data.
+
+Syntax Example::
+
+  dce.stub_data; content:"%|00|C|00|O|00|M|00|S|00|P|00|E|00|C|00|%";
```
Comment (victorjulien) [inline doc/userguide/rules/dcerpc-keywords.rst]: dcerpc.stub_data

File: doc/userguide/rules/dcerpc-keywords.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+Signature Example:
+
+.. container:: example-rule
+
+  alert tcp any any -> $HOME_NET any (msg:"TGI LATERAL DCE/RPC stub data contains PSEXEC"; :example-rule-emphasis:`dcerpc.stub_data; content:"P|00|S|00|E|00|X|00|E|00|S|00|V|00|C";` sid:1;)
```
Comment (victorjulien) [inline doc/userguide/rules/dcerpc-keywords.rst]: Also missing reference to old keyword name here at the end of this section.

File: src/detect-dce-iface.c [hunk 5]
```diff
@@ -379,7 +381,7 @@ static int DetectDceIfaceSetup(DetectEngineCtx *de_ctx, Signature *s, const char
 {
     DetectDceIfaceData *did = DetectDceIfaceArgParse(arg);
     if (did == NULL) {
-        SCLogError(SC_ERR_INVALID_SIGNATURE, "Error parsing dec_iface option in "
+        SCLogError(SC_ERR_INVALID_SIGNATURE, "Error parsing dce_iface option in "
```
Comment (victorjulien) [inline src/detect-dce-iface.c]: dcerpc.iface

File: src/detect-dce-opnum.c [hunk 6]
```diff
@@ -71,6 +71,8 @@ void DetectDceOpnumRegister(void)
 {
     sigmatch_table[DETECT_DCE_OPNUM].name = "dcerpc.opnum";
     sigmatch_table[DETECT_DCE_OPNUM].alias = "dce_opnum";
+    sigmatch_table[DETECT_DCE_OPNUM].desc = "match dcerpc operation number";
```
Comment (victorjulien) [inline src/detect-dce-opnum.c]: should we use the DCE/RPC notation here?

File: src/detect-dce-stub-data.c [hunk 7]
```diff
@@ -123,6 +123,8 @@ void DetectDceStubDataRegister(void)
 {
     sigmatch_table[DETECT_DCE_STUB_DATA].name = "dcerpc.stub_data";
     sigmatch_table[DETECT_DCE_STUB_DATA].alias = "dce_stub_data";
+    sigmatch_table[DETECT_DCE_STUB_DATA].desc = "sticky buffer to match dcerpc stub data ";
```
Comment (victorjulien) [inline src/detect-dce-stub-data.c]: should we use the DCE/RPC notation here?

**Pull request closed without merging**